nifi-extension-bundles/nifi-gitlab-bundle/nifi-gitlab-extensions/src/main/java/org/apache/nifi/gitlab/GitLabAuthenticationType.java

```diff
@@ -0,0 +1,22 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.nifi.gitlab;
+
+public enum GitLabAuthenticationType {
+    ACCESS_TOKEN
+}
```

Comment:
Does it make sense to turn this into a `DescribedValue` similar to the existing `GitHubAuthenticationType`? 

```suggestion
public enum GitLabAuthenticationType implements DescribedValue {
    ACCESS_TOKEN("Access Token", "Group, Project, or Personal Access Token");

    private final String displayName;
    private final String description;

    GitLabAuthenticationType(final String displayName, final String description) {
        this.displayName = displayName;
        this.description = description;
    }

    @Override
    public String getValue() {
        return name();
    }

    @Override
    public String getDisplayName() {
        return displayName;
    }

    @Override
    public String getDescription() {
        return description;
    }
}
```

If so, we could provide users with more readable names / descriptions and apply minor simplifications in property declaration, see other comments.

Reply:
Yes good idea, this makes sense, I'll make these changes related to use of `DescribedValue`.